packages/fuels-core/src/types/transaction_builders/script_dry_runner.rs

```diff
@@ -0,0 +1,138 @@
+};
+use itertools::Itertools;
+
+use super::dry_runner::DryRun;
+use crate::{
+    constants::WITNESS_STATIC_SIZE,
+    types::{errors::Result, transaction_builders::DryRunner, AssetId},
+};
+
+pub(crate) struct ScriptDryRunner<R> {
+    dry_runner: R,
+    cache: HashMap<fuel_tx::Script, DryRun>,
+    num_total_witnesses: usize,
+}
+
+impl<R> ScriptDryRunner<R> {
+    pub fn new(dry_runner: R, num_total_witnesses: usize) -> Self {
+        Self {
+            dry_runner,
+            cache: HashMap::new(),
+            num_total_witnesses,
+        }
+    }
+}
+
+impl<R: DryRunner> ScriptDryRunner<R> {
+    pub async fn run(&mut self, mut tx: fuel_tx::Script) -> Result<DryRun> {
+        self.add_fake_witnesses(&mut tx);
+        self.add_fake_coins(&mut tx);
+        self.saturate_with_variable_outputs(&mut tx);
+        self.set_script_gas_limit_to_max(&mut tx);
+
+        self.cached_or_run(tx).await
+    }
+
+    fn set_script_gas_limit_to_max(&mut self, tx: &mut fuel_tx::Script) {
+        let consensus_params = self.dry_runner.consensus_parameters();
+        let max_gas = tx.max_gas(consensus_params.gas_costs(), consensus_params.fee_params()) + 1;
+        *tx.script_gas_limit_mut() = consensus_params.tx_params().max_gas_per_tx() - max_gas;
+    }
+
+    fn saturate_with_variable_outputs(&mut self, tx: &mut fuel_tx::Script) {
+        let consensus_params = self.dry_runner.consensus_parameters();
+
+        let max_outputs = usize::from(consensus_params.tx_params().max_outputs());
+        let used_outputs = tx.outputs().len();
+
+        let unused_outputs = max_outputs.saturating_sub(used_outputs);
+        super::add_variable_outputs(tx, unused_outputs);
+    }
+
+    // When dry running a tx with `utxo_validation` off, the node will not validate signatures.
+    // However, the node will check if the right number of witnesses is present.
+    // This function will create witnesses from a default `Signature` such that the total length matches the expected one.
+    // Using a `Signature` ensures that the calculated fee includes the fee generated by the witnesses.
+    fn add_fake_witnesses(&mut self, tx: &mut fuel_tx::Script) {
+        let witness: Witness = Signature::default().as_ref().into();
+        let dry_run_witnesses: Vec<_> = repeat(witness).take(self.num_total_witnesses).collect();
+
+        *tx.witnesses_mut() = dry_run_witnesses;
+    }
+
+    fn add_fake_coins(&mut self, tx: &mut fuel_tx::Script) {
+        let consensus_params = self.dry_runner.consensus_parameters();
+        if let Some(fake_input) =
+            Self::needs_fake_base_input(tx.inputs(), consensus_params.base_asset_id())
+        {
+            tx.inputs_mut().push(fake_input);
+
+            // Add an empty `Witness` for the `coin_signed` we just added
+            tx.witnesses_mut().push(Witness::default());
+            tx.set_witness_limit(tx.witness_limit() + WITNESS_STATIC_SIZE as u64);
+        }
+    }
+
+    fn needs_fake_base_input(
```

Comment:
There exists a tx validity condition that requires spendable inputs to be present:
https://github.com/FuelLabs/fuel-vm/blob/898789cc58ce549a92550d680df96fa0d0480002/fuel-tx/src/transaction/validity/error.rs#L15

I didn't come across anything that indicates that there needs to be base asset inputs. Running the tests with the base asset id check removed, seems to work fine.

Reply:
This is inherited. @hal3e do you remember the original reason why we've added the base asset req?

Reply:
@MujkicA I've tested it and it seems you're right, dry runs are passing without any base assets in inputs.

Something must have changed in the meantime @hal3e . 

In any case I've removed the requirement of the asset being of the base id.